### ISSUE TITLE
msat fields are integers instead of strings now in CLN master

### ISSUE THREAD

**Issue by grubles** [ElementsProject/peerswap#181]
```
2023-03-20T00:38:07.122Z INFO    plugin-peerswap: 20:38:07 clightning.go:431: getinfo err json: cannot 
unmarshal number into Go struct field NodeInfo.fees_collected_msat of type string                      
2023-03-20T00:38:07.123Z INFO    plugin-peerswap: Killing plugin: exited before replying to init
```

Looks like https://github.com/ElementsProject/lightning/pull/5986 in CLN master affects Peerswap.

Can we support both types of fields?

**Comment by nepet** [comment]
It is on my plate to add a generator for go bindings using core-lightning's `msggen`. This way we would have bindings that reflect cln's internal api. I am not sure how broad the version support for the generated bindings would be, bindings could set deprecated fields as optional, but in any case it is on the plugin to handle different versions of core-lightning and the expected commands and returns. I don't know if this will be ready for `v23.05`.

In the meantime we have to update `glightning` to support v23.05. We can then use `replace` to import two versions of the bindings and peerswap then has to check on cln and use the according version of the bindings. This might become ugly but I guess this is the easiest way to handle it for the moment, allowing support of `v23.05` as well as older versions.

**Comment by wtogami** [comment]
> We can then use go mods replace to import two versions of the bindings

Is that the best way to make it maintainable? Thereafter every fix to the bindings would need to committed twice. It seems more more self contained for the bindings to understand both output formats until the old versions are totally retired?

**Comment by grubles** [comment]
Fixed in #183
